solution-base/mongodb/how_to_upgrade.md

```diff
@@ -0,0 +1,19 @@
+# MongoDB Sharded Upgrade Process
+
+## Overview
+
+Upgrading MongoDB sharded involves several steps to ensure a smooth transition to the new version. Below is a structured guide to follow for a successful upgrade (note that all along
+I will be taking as an example the upgrade from 5 to 6):
+
+## Upgrade steps
+
+
+1. Check for all available versions of the chart using `helm search repo bitnami/mongodb-sharded --versions` to check the configuration values for a specified version: `helm show values --version 6.6.7  bitnami/mongodb-sharded`
+2. Bump mongo version in `.github/scripts/end2end/kustomization.yaml` and `solution-base/deps.yaml` files.
+3. Upgrade mongodb sharded version in charts => `CHART_MONGO_SHARDED_VERSION` in the `solution-base/mongodb/Makefile` file (e.g. `CHART_MONGO_SHARDED_VERSION:="6.6.7"`)
+4. Upgrade charts to the version targetted => `make fetch-mongodb-sharded`
+5. Manually update the patches by applying the changes manually in the new upgraded charts (note that this can be applied only to patches that target mongo sharded) =>  `git diff -- solution-base/mongodb/charts/mongodb-sharded/values.yaml > solution-base/mongodb/patches/secret-name.patch` (this operation needs to be done for every patch).
+6. Once the patches are updated, apply them to the charts with the command `make patch`.
+7. Applying the changes introduced by the upgrade can then be done (by checking the release notes for the new version e.g `https://www.mongodb.com/docs/v6.0/release-notes/6.0/`)
```

Comment:
i don't understand what you mean withi this step. You already "applied the changes" earlier...

Reply:
i am talking about the additional changes ( like in the case of mongo 6 the switch from mongo to mongosh that is not done directly on the charts/patches)

Reply:
please reword to make this easier to understand

Reply:
https://github.com/scality/Zenko/pull/2028/commits/4809ce468c3808ca839a6924c79b0931dcd09e09#diff-a2232853b265dbe8c46e493c507a5b212f5e815cd22ab84e8a526ad808d3dab8R17